### PR TITLE
[REG2.055] Issue 12422 - Templated nested function is inferred as `pure` even if it calls impure functions

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=12422

### CODE REVIEW

File: src/expression.c [hunk 1]
```diff
@@ -2397,6 +2397,10 @@ void Expression::checkPurity(Scope *sc, FuncDeclaration *f)
     {
         // The attributes of outerfunc should be inferred from the call of f.
     }
+    else if (f->isInstantiated())
+    {
+        // The attributes of f is inferred from its body.
```
Comment (WalterBright) [inline src/expression.c]: s/is/are/

File: src/expression.c [hunk 2]
```diff
@@ -2397,6 +2397,10 @@ void Expression::checkPurity(Scope *sc, FuncDeclaration *f)
...
+    }
     else if (f->isFuncLiteralDeclaration())
     {
         // The attributes of f is always inferred in its declared place.
```
Comment (WalterBright) [inline src/expression.c]: might as will fix this line, too: s/is/are/

**Comment by WalterBright** [comment]
Merged despite typos because it's a regression.

**Comment by 9rnsr** [comment]
Sorry for the late reply. In #4447, I also fixed the typos.

**Comment by MartinNowak** [comment]
Can you guys please mark regressions with the 2.067 milestone, so that I have a chance to merge the pick pulls ;).

**Comment by 9rnsr** [comment]
OK, I checked the difference between master and 2.067 branches, and found two PRs which need to be marked with the 2.067 milestone. 
https://github.com/D-Programming-Language/dmd/pull/4417
https://github.com/D-Programming-Language/dmd/pull/4446

**Comment by MartinNowak** [comment]
Thanks @9rnsr, I indeed missed those.

**Pull request merged**